### ISSUE TITLE
New panels: ColumnMetadataTable & RowMetadataTable

### ISSUE THREAD

**Issue by kevinrue** [iSEE/iSEEu#21]
## Issue

In relation to #3 

There could be another two types of panels, namely `ColumnMetadataTable` and `RowMetadataTable`, that display columns in `rowData()` and `colData()` as rows, and (optional) metadata about those as columns.

This would make it possible to select colData and rowData columns in those tables, and other panels (possibly all panels using the new dynamic selection infrastructure) could listen on that selection.

A typical application would be to select a colData column, and apply it as a the color to all other panels.

## Implementation:

DataFrames contains an `elementMetadata` component that is either `NULL` or a `DataFrame` with as many rows as there are columns in the main `DataFrame`.

Basically, all we would have to do, if the users didn't populate `elementMetadata`, would be to create a 0-column DataFrame as below.
We might have to add a dummy column to actually show a table in the panel , though.

```
> elementMetadata(colData(sce))
NULL
> elementMetadata(colData(sce)) <- DataFrame(row.names = colnames(colData(sce)))
> elementMetadata(colData(sce))
DataFrame with 23 rows and 0 columns
```

**Comment by LTLA** [comment]
To curb some enthusiasm: this will not be easy. There is no infrastructure for panels to indicate that they are open to receiving a choice of column/row metadata variable. This would need to be built from scratch. While one could use the proposed panels to forcibly `updateSelectInput` all potential receiving slots, that would be terribly impolite and so we're not going to do that.

The multi-panel solution proposed above is also rather unappealing in the sense that the selector panels have no function than to transmit to other panels. All other panels (excepting the `GeneSetTable`, but I have plans for that) aim to be useful in and of themselves, otherwise the general design principle of having independent and interchangeable panels doesn't hold.

The least-worst of all approaches would be for the mothership to implement another dropdown menu allowing users to forcibly override eligible slots. This would require all panels to declare which column/row metadata variables are permissible in which slots... and there are quite a lot of them.

**Comment by kevinrue** [comment]
The idea was fun (well not) during the 4 years that it lasted
